lib/virtual_regional_office/client.rb

```diff
@@ -7,35 +7,54 @@ class Client < Common::Client::Base
     include Common::Client::Concerns::Monitoring
     configuration VirtualRegionalOffice::Configuration
 
-    # Initializes the MAS client.
+    # Initializes the VRO client.
     #
     # @example
     #
-    # VirtualRegionalOffice::Client.new({
-    #  "veteranIcn": "9000682",
-    #  "diagnosticCode": "7101",
-    #  "claimSubmissionId": "1234"
-    # })
+    # VirtualRegionalOffice::Client.new(diagnostic_code: 7101, claim_submission_id: 1234)
     def initialize(params)
-      @veteran_icn = params[:veteran_icn]
-      @diagnostic_code = params[:diagnostic_code]
-      @claim_submission_id = params[:claim_submission_id]
+      @diagnostic_code = params[:diagnostic_code].to_s
+      @claim_submission_id = params[:claim_submission_id].to_s
 
-      raise ArgumentError, 'no veteran_icn passed in for request.' if @veteran_icn.blank?
       raise ArgumentError, 'no diagnostic_code passed in for request.' if @diagnostic_code.blank?
       raise ArgumentError, 'no claim_submission_id passed in for request.' if @claim_submission_id.blank?
     end
 
-    def assess_claim
+    def assess_claim(veteran_icn:)
       params = {
-        veteranIcn: @veteran_icn,
+        veteranIcn: veteran_icn,
         diagnosticCode: @diagnostic_code,
         claimSubmissionId: @claim_submission_id
       }
 
       perform(:post, Settings.virtual_regional_office.health_assessment_path, params.to_json.to_s, headers_hash)
     end
 
+    def generate_summary(veteran_info:, evidence:)
+      params = {
+        claimSubmissionId: @claim_submission_id,
+        diagnosticCode: @diagnostic_code,
+        veteranInfo: veteran_info,
+        evidence: evidence
+      }
+
+      perform(:post, Settings.virtual_regional_office.evidence_pdf_path, params.to_json.to_s, headers_hash)
+    end
+
+    def download_summary
+      path = "#{Settings.virtual_regional_office.evidence_pdf_path}/#{@claim_submission_id}"
+      perform(:get, path, {}, headers_hash.merge(Accept: 'application/pdf'))
+    end
+
+    # Tiny middleware to replace the configuration option `faraday.response :json` with behavior
+    # that only decodes JSON for application/json responses. This allows us to handle non-JSON
+    # responses (e.g. application/pdf) without loss of convenience.
+    def perform(method, path, params, headers = nil, options = nil)
+      result = super(method, path, params, headers, options)
```

Comment:
Wondering if this will work:
```suggestion
      result = super
```
https://www.rubyguides.com/2018/09/ruby-super-keyword/:
> It will pass along the arguments used for the original method call to the new one, including [keyword arguments](https://www.rubyguides.com/2018/06/rubys-method-arguments/) & a block if given.